### ISSUE TITLE
uWSGI does not start

### ISSUE THREAD

**Issue by skorokithakis** [piku/piku#204]
I just installed Piku on the latest Raspbian, but deploying a simple Flask app fails with:

```
The -s/--socket option is missing and stdin is not a socket.
```

The Procfile is:
```
wsgi: app:app
```

And the file is called `app.py`.

**Comment by rcarmo** [comment]
Can you tell us how you installed, try the sample app and post the full terminal output? Given #202 you might be missing more packages.

**Comment by skorokithakis** [comment]
I installed Raspbian onto the SD card with the official imager, SSHed in and ran the bootstrap installer, as per the instructions.

Here's the output of the sample app:

```
worker.1 | 7
wsgi.1   | *** Starting uWSGI 2.0.18-debian (32bit) on [Tue Aug 31 01:09:17 2021] ***
worker.1 | 8
wsgi.1   | compiled with version: 8.2.0 on 10 February 2019 02:42:46
wsgi.1   | os: Linux-5.10.17-v7+ #1414 SMP Fri Apr 30 13:18:35 BST 2021
wsgi.1   | nodename: raspberrypi
wsgi.1   | machine: armv7l
wsgi.1   | clock source: unix
wsgi.1   | pcre jit disabled
wsgi.1   | detected number of CPU cores: 4
wsgi.1   | current working directory: /home/piku/.piku/uwsgi-enabled
wsgi.1   | detected binary path: /usr/bin/uwsgi-core
wsgi.1   | chdir() to /home/piku/.piku/apps/sample_python_app
wsgi.1   | your processes number limit is 6873
wsgi.1   | your memory page size is 4096 bytes
wsgi.1   | detected max file descriptor number: 1024
wsgi.1   | lock engine: pthread robust mutexes
wsgi.1   | thunder lock: disabled (you can enable it with --thunder-lock)
wsgi.1   | Python version: 3.7.3 (default, Jan 22 2021, 20:04:44)  [GCC 8.3.0]
wsgi.1   | Set PythonHome to /home/piku/.piku/envs/sample_python_app
wsgi.1   | Python main interpreter initialized at 0xa96a18
wsgi.1   | python threads support enabled
wsgi.1   | The -s/--socket option is missing and stdin is not a socket.
worker.1 | 9
worker.1 | [uwsgi-daemons] respawning "python worker.py" (uid: 1002 gid: 33)
worker.1 | 0
worker.1 | 1
```

**Comment by skorokithakis** [comment]
Any ideas? Piku is currently unusable for me with this.

**Comment by namsral** [comment]
I had the same issue on Ubuntu 20.04.

The uwsgi configuration generated by [piku.py](https://github.com/piku/piku/blob/master/piku.py) is incompatible with the installed uwsgi package, syntax of the http socket option did not match.

See `uwsgi --help` or `man uwsgi-core` in your setup for the correct http socket syntax and update [piku.py#L992](https://github.com/piku/piku/blob/master/piku.py#L992) accordingly.

**Comment by rcarmo** [comment]
Hmmm. We need to get a fix for that in then. I missed that in my recent deployments since I'm mostly doing queue processing and no web servers, but if it is now breaking in 20.04 and Buster it needs to be fixed.

**Comment by skorokithakis** [comment]
Also, it would generally be a good idea if Piku pinned minor versions of packages, so things wouldn't change from under you.

**Comment by skorokithakis** [comment]
Hmm, it looks like Piku uses the system uWSGI package, so pinning will be rather hard... It would be easier to control its dependencies (and thus help stability) if it used its own uWSGI, but it might just be easier to fix the socket thing now.

**Comment by rcarmo** [comment]
Yes, `piku` only uses system packages (except for click) by design. `uWSGI` is not a trivial dependency, but since we target Ubuntu LTS and Debian the packages change very seldom (this is the first time I recall having a "change" in `uWSGI` that wasn't a new feature).

**Comment by rcarmo** [comment]
@namsral, can you do a pull request with the working syntax for that line, please? I am not in a bandwidth friendly location and don't have my own laptop.

**Comment by namsral** [comment]
@rcarmo, see https://github.com/piku/piku/pull/205 to fix this issue.

**Comment by namsral** [comment]
I'm boggled by the configuration inconsistency between the pip and Debian system package.

pip package:

```sh
$ uwsgi --version
2.0.19.1
$ uwsgi --help
    --http-use-socket                       forward request to the specified uwsgi socket
```

Ubuntu system package:

```sh
$ uwsgi --version
2.0.18-debian
$ uwsgi --help
    --http-socket                         bind to the specified UNIX/TCP socket using HTTP protocol
```

**Comment by rcarmo** [comment]
Yes, it is listed since 2016. That file is not applicable anymore since `piku` is not designed to use a `pip` installed `uwsgi`.

**Comment by rcarmo** [comment]
I've since removed `requirements.txt` and moved it to `requirements-dev.txt` to avoid this kind of confusion. Will revise  instructions later.

**Comment by chr15m** [comment]
@rcarmo this is reported as fixed on #196 so we can probably close this.
